go/exosecurity/test-fixtures/complex.go

```diff
@@ -0,0 +1,43 @@
+package exosecurityTestFixtures
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/Originate/exocom/go/exosecurity"
+	"github.com/Originate/exocom/go/structs"
+)
+
+// ComplexTestFixture is a complex test fixture
+type ComplexTestFixture struct{}
+
+// GetMessageValidator returns a message validator
+func (s *ComplexTestFixture) GetMessageValidator() exosecurity.MessageAuthorizer {
+	return func(message *structs.Message, requestData exosecurity.DataRequester) bool {
+		if message.Name == "create user" {
+			receivedMessage := requestData(&structs.Message{
+				Name:       "retrieve user session",
+				Payload:    message.Auth,
+				ActivityID: "456",
+				ID:         "123",
+			})
+			bytes, err := json.Marshal(receivedMessage.Payload)
+			if err != nil {
+				fmt.Println(err)
+				return false
+			}
+			payload := map[string]interface{}{}
+			err = json.Unmarshal(bytes, &payload)
+			if err != nil {
+				fmt.Println(err)
+				return false
+			}
```

Comment:
The marshalling and unmarshalling the payload in order to return the should be built into exosecurity. The requester function should not return the "security response" message, but it payload (as a message)